src/cli.rs

```diff
@@ -179,6 +179,14 @@ pub enum SubCommand {
     /// Generate a Vector configuration containing a list of components.
     Generate(generate::Opts),
 
+    /// Generate the configuration schema for this version of Vector.
+    ///
+    /// A JSON Schema document will be written to stdout that represents the valid schema for a
+    /// Vector configuration. This schema is based on the "full" configuration, such that for usages
+    /// where a configuration is split into multiple files, the schema would apply to those files
+    /// only when concatentated together.
+    GenerateSchema,
```

Comment:
👍 

Could we label this as experimental so users will expect some breaking changes while we stabilize this feature? Could also mark it as hidden for now.

Reply:
Sure, I'll add that to the command description.

Reply:
Fixed in 8b55d4493.